graphite/graphite_tcp.go

```diff
@@ -0,0 +1,81 @@
+package graphite
+
+import (
+	"bufio"
+	"log"
+	"net"
+	"strings"
+)
+
+// TCPServer processes Graphite data received over TCP connections.
+type TCPServer struct {
+	writer SeriesWriter
+	parser *Parser
+
+	Database string
+}
+
+// NewTCPServer returns a new instance of a TCPServer.
+func NewTCPServer(p *Parser, w SeriesWriter) *TCPServer {
+	t := TCPServer{
+		parser: p,
+		writer: w,
+	}
+	return &t
```

Comment:
Can you have this directly return the new instance instead of assigning it to `t` first?